src/main/controllers/UserOptionController.ts

```diff
@@ -1,31 +1,64 @@
 import { AuthedRequest } from '../interfaces/AuthedRequest';
 import { Response } from 'express';
-import { hasProperty } from '../utils/utils';
-import { MISSING_OPTION_ERROR } from '../utils/error';
+import { constructOptionsStringFromArray, hasProperty, isObjectEmpty } from '../utils/utils';
+import { selectOptionError } from '../utils/error';
 import { ADD_USER_URL, GENERATE_REPORT_URL, MANAGER_USER_URL } from '../utils/urls';
 import { RootController } from './RootController';
 import autobind from 'autobind-decorator';
+import { UserOption } from '../utils/UserOption';
+import { SelectItem } from '../interfaces/SelectItem';
+import { BETA_ADD, GAMMA_GENERATE_REPORT } from '../app/feature-flags/flags';
 
 @autobind
 export class UserOptionController extends RootController {
-  public get(req: AuthedRequest, res: Response) {
-    return super.get(req, res,'user-option');
+  public async get(req: AuthedRequest, res: Response) {
+    const options = await this.getOptionsForSelect();
+    return super.get(req, res,'user-option', { content: { options: options } });
   }
 
-  public post(req: AuthedRequest, res: Response) {
+  public async post(req: AuthedRequest, res: Response) {
     if (!hasProperty(req.body, 'userAction')) {
-      return super.post(req, res, 'user-option', { error: {
-        userAction: { message: MISSING_OPTION_ERROR }
-      }});
+      const options = await this.getOptionsForSelect();
+      return super.post(req, res, 'user-option', {
+        content: { options: options },
+        error: { userAction: { message: selectOptionError(this.composeOptions(options)) } }
+      });
     }
 
     switch (req.body.userAction) {
-      case 'manage-user':
+      case UserOption.MANAGE_USER:
         return res.redirect(MANAGER_USER_URL);
-      case 'add-user':
+      case UserOption.ADD_USER:
         return res.redirect(ADD_USER_URL);
-      case 'generate-report':
+      case UserOption.GENERATE_REPORT:
         return res.redirect(GENERATE_REPORT_URL);
     }
   }
+
+  private async getOptionsForSelect(): Promise<SelectItem[]> {
+    const items = [{value: UserOption.MANAGE_USER, text: 'Manage an existing user'}];
+
+    if (this.getFeatureFlags()) {
+      const featureFlags = await this.getFeatureFlags().getAllFlagValues();
+
+      if (!isObjectEmpty(featureFlags)) {
```

Comment:
I'm not sure if this extra check is necessary as 
`featureFlags[BETA_ADD]`
will not meet the condition if it's undefined (if featureFlags is empty)